python/lsst/dax/distribution/DataIngest.py

```diff
@@ -0,0 +1,340 @@
+
+        Parameters
+        ----------
+        db_name : str
+            Name of the database to be published.
+        """
+        success = False
+        cmd = 'ingest/database/' + db_name
+        success, status, r_json = self._putToIngest(cmd, {'auth_key':self._auth_key})
+        if not success:
+            print("ERROR publishing", db_name, "status=", status, "r_json=", r_json)
+        return success, status, r_json
+
+class IngestTransaction():
+    """RAII object to make sure transactions are closed.
+    Throws RunTimeError if transaction cannot be started or closed.
+    self.abort needs to be set to False if the elements of the
+    transaction are successful before __exit__ is called.
+    """
+
+    def __init__(self, data_ingest, db_name):
+        self._data_ingest = data_ingest
+        self._db_name = db_name
+        self._id = -1
+        self.abort = True
+
+    def __repr__(self):
+        out = 'data_ingest(' + str(self._data_ingest) + ") db=" + self._db_name + " id=" + str(self._id)
+        return out
+
+    def __enter__(self):
+        success, id = self._data_ingest.startTransaction(self._db_name)
+        if success:
+            self._id = id
+            print('Transaction started ', self._db_name, self._id)
+        else:
+            raise RuntimeError('Transaction failed to start ' + self._db_name + self._id)
+        return self._id
+
+    def __exit__(self, e_type, e_value, e_traceback):
+        success = False
+        if e_type:
+            print("__exit__ exception=", e_type, "val=", e_value, "trace=", e_traceback)
+            return False
+        content = None
+        status = -1
+        if self._id > -1:
+            success, status, content = self._data_ingest.endTransaction(self._id, abort=False)
+        if not success:
+            print("ERROR Transaction end failed ", self._db_name, self._id, status, content)
+            raise RuntimeError('Transaction failed ' + self._db_name + " trans_id="+ str(self._id)
+                                + " status=" + str(status) + " content=" + str(content))
+        return True
+
+
+if __name__ == "__main__":
+    ingest = DataIngest('localhost', 25080)
+    # No point in continuing if the ingest system can't be contacted.
+    if not ingest.isIngestAlive():
+        print("ERROR ingest server not responding.")
+        exit(1)
+    # Try sending the database. This will fail if the database already exists.
+    if not ingest.sendDatabase("fakeIngestCfgsTest/test102.json"):
+        print("ERROR failed to send database configuration to ingest.")
+    # Ingest the test Object tables schema. This will fail if the
+    # database already exists.
+    if not ingest.sendTableSchema("fakeIngestCfgsTest/test102_Object.json"):
+        print("ERROR failed to send Object table schema")
+    # Start a transaction
+    i_transaction = IngestTransaction(ingest, 'test102')
+    transaction_status = None
+    try:
+        with i_transaction as t_id:
```

Comment:
I wound up not using this code outside of testing. If there's reliability problems with clients, it will be worth having the server close the individual transactions so that the server really knows which files have been ingested. I'm hoping client reliability wont be that much of an issue.